examples/simple_intersection/example_intersection.cpp

```diff
@@ -0,0 +1,115 @@
+//
+//  o    o   o   o   j-2
+//
+
+int main()
+{
+  Kokkos::initialize();
+  {
+    using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+    using MemorySpace = typename ExecutionSpace::memory_space;
+    ExecutionSpace execution_space;
+
+    std::cout << "Create grid with bounding boxes" << '\n';
+    float Lx = 100.0;
+    float Ly = 100.0;
+    float Lz = 100.0;
+    int nx = 11;
+    int ny = 11;
+    int nz = 11;
+    int n = nx * ny * nz;
+    float hx = Lx / (nx - 1);
+    float hy = Ly / (ny - 1);
+    float hz = Lz / (nz - 1);
+
+    std::function<int(int, int, int)> ind = [nx, ny](int i, int j, int k) {
+      return i + j * nx + k * (nx * ny);
+    };
+
+    Kokkos::View<ArborX::Box *, MemorySpace> bounding_boxes("bounding_boxes",
+                                                            n);
+    auto bounding_boxes_host = Kokkos::create_mirror_view(bounding_boxes);
+
+    for (int i = 0; i < nx; ++i)
+      for (int j = 0; j < ny; ++j)
+        for (int k = 0; k < nz; ++k)
+        {
+          ArborX::Point p_lower{
+              {(i - .25) * hx, (j - .25) * hy, (k - .25) * hz}};
+          ArborX::Point p_upper{
+              {(i + .25) * hx, (j + .25) * hy, (k + .25) * hz}};
+          bounding_boxes_host[ind(i, j, k)] = {p_lower, p_upper};
+        }
+    Kokkos::deep_copy(bounding_boxes, bounding_boxes_host);
+    std::cout << "Bounding boxes set up." << '\n';
+
+    std::cout << "Creating BVH tree." << '\n';
+    ArborX::BVH<MemorySpace> const tree(execution_space, bounding_boxes);
+    std::cout << "BVH tree set up." << '\n';
+
+    std::cout << "Filling queries." << '\n';
+    Kokkos::View<decltype(ArborX::intersects(ArborX::Box{})) *, MemorySpace>
+        queries("queries", n);
+    Kokkos::parallel_for(
+        "fill_queries",
+        Kokkos::RangePolicy<ExecutionSpace>(execution_space, 0, n),
+        KOKKOS_LAMBDA(int i) {
+          queries(i) = ArborX::intersects(bounding_boxes(i));
+        });
+    std::cout << "Queries set up." << '\n';
+
+    std::cout << "Starting the queries." << '\n';
+    // The query will resize indices and offsets accordingly
+    Kokkos::View<int *, MemorySpace> indices("indices", 0);
+    Kokkos::View<int *, MemorySpace> offsets("offsets", 0);
+
+    ArborX::query(tree, execution_space, queries, indices, offsets);
+    std::cout << "Queries done." << '\n';
+
+    std::cout << "Starting checking results." << '\n';
+    auto offsets_host =
+        Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, offsets);
+    auto indices_host =
+        Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, indices);
+
+    assert(static_cast<int>(offsets_host.size()) == n + 1);
+    for (int i = 0; i < n + 1; ++i)
+    {
+      assert(offsets_host(i) == i);
+    }
+    assert(static_cast<int>(indices_host.size()) == n);
+    for (int i = 0; i < n; ++i)
+    {
+      assert(indices_host(i) == i);
+    }
+    std::cout << "Checking results successful." << '\n';
+  }
```

Comment:
Maybe replace asserts with a boolean flag, and only indicate at the end whether it failed.